.githooks/pre-commit

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+echo "Running static analysis with checkstyle and ktlint..."
+
+./gradlew apiDump
```

Comment:
I don't think we want to automatically run this. It should be something that's intentional so that we don't change our public API inadvertently. Most of our changes shouldn't be impacted by this.